File: sample-videochat-conference/src/main/java/com/quickblox/sample/conference/fragments/ConversationFragmentCallbackListener.java
package com.quickblox.sample.conference.fragments;

import com.quickblox.conference.ConferenceSession;
import com.quickblox.sample.conference.activities.CallActivity;
import com.quickblox.videochat.webrtc.callbacks.QBRTCSessionStateCallback;

import org.webrtc.CameraVideoCapturer;


public interface ConversationFragmentCallbackListener {

    void addClientConnectionCallback(QBRTCSessionStateCallback<ConferenceSession> clientConnectionCallbacks);

    void removeClientConnectionCallback(QBRTCSessionStateCallback clientConnectionCallbacks);

    void addCurrentCallStateCallback(CallActivity.CurrentCallStateCallback currentCallStateCallback);

    void removeCurrentCallStateCallback(CallActivity.CurrentCallStateCallback currentCallStateCallback);

    void addOnChangeDynamicToggle(CallActivity.OnChangeDynamicToggle onChangeDynamicCallback);

    void removeOnChangeDynamicToggle(CallActivity.OnChangeDynamicToggle onChangeDynamicCallback);

    void onSetAudioEnabled(boolean isAudioEnabled);

    void onSetVideoEnabled(boolean isNeedEnableCam);

    void onSwitchAudio();

    void onLeaveCurrentSession();

    void onSwitchCamera(CameraVideoCapturer.CameraSwitchHandler cameraSwitchHandler);

    void onStartJoinConference();
}

File: sample-videochat-conference-kotlin/src/main/java/com/quickblox/sample/videochat/conference/kotlin/fragments/ConversationFragmentCallbackListener.kt
package com.quickblox.sample.videochat.conference.kotlin.fragments

import com.quickblox.conference.ConferenceSession
import com.quickblox.sample.videochat.conference.kotlin.activities.CallActivity
import com.quickblox.videochat.webrtc.callbacks.QBRTCSessionStateCallback
import org.webrtc.CameraVideoCapturer


interface ConversationFragmentCallbackListener {

    fun addClientConnectionCallback(clientConnectionCallbacks: QBRTCSessionStateCallback<ConferenceSession>)

    fun removeClientConnectionCallback(clientConnectionCallbacks: QBRTCSessionStateCallback<*>)

    fun addCurrentCallStateCallback(currentCallStateCallback: CallActivity.CurrentCallStateCallback)

    fun removeCurrentCallStateCallback(currentCallStateCallback: CallActivity.CurrentCallStateCallback)

    fun addOnChangeDynamicToggle(onChangeDynamicCallback: CallActivity.OnChangeDynamicToggle)

    fun removeOnChangeDynamicToggle(onChangeDynamicCallback: CallActivity.OnChangeDynamicToggle)

    fun onSetAudioEnabled(isAudioEnabled: Boolean)

    fun onSetVideoEnabled(isNeedEnableCam: Boolean)

    fun onSwitchAudio()

    fun onLeaveCurrentSession()

    fun onSwitchCamera(cameraSwitchHandler: CameraVideoCapturer.CameraSwitchHandler)

    fun onStartJoinConference()
}